modules/kameleonvolume/rendering/renderablekameleonvolume.cpp

```diff
@@ -0,0 +1,375 @@
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/kameleonvolume/rendering/renderablekameleonvolume.h>
+#include <modules/kameleonvolume/kameleonvolumereader.h>
+
+#include <openspace/rendering/renderable.h>
+#include <openspace/engine/openspaceengine.h>
+#include <openspace/rendering/renderengine.h>
+#include <openspace/rendering/raycastermanager.h>
+#include <glm/glm.hpp>
+#include <glm/gtc/matrix_transform.hpp>
+
+#include <ghoul/opengl/ghoul_gl.h>
+#include <ghoul/filesystem/filesystem.h>
+#include <ghoul/filesystem/cachemanager.h>
+#include <ghoul/logging/logmanager.h>
+
+#include <modules/volume/rawvolumereader.h>
+#include <modules/volume/rawvolumewriter.h>
+#include <modules/volume/rawvolume.h>
+
+namespace {
+    const std::string _loggerCat = "RenderableKameleonVolume";
+}
+
+namespace openspace {
+
+RenderableKameleonVolume::RenderableKameleonVolume(const ghoul::Dictionary& dictionary)
+    : Renderable(dictionary)
+    , _dimensions("dimensions", "Dimensions")
+    , _variable("variable", "Variable")
+    , _lowerDomainBound("lowerDomainBound", "Lower Domain Bound")
+    , _upperDomainBound("upperDomainBound", "Upper Domain Bound")
+    , _domainScale("domainScale", "Domain scale")
+    , _autoDomainBounds(false)
+    , _lowerValueBound("lowerValueBound", "Lower Value Bound", 0.0, 0.0, 1)
+    , _upperValueBound("upperValueBound", "Upper Value Bound", 1, 0.01, 1)
+    , _autoValueBounds(false)
+    , _gridType("gridType", "Grid Type", properties::OptionProperty::DisplayType::Dropdown)
+    , _autoGridType(false)
+    , _clipPlanes(nullptr)
+    , _stepSize("stepSize", "Step Size", 0.02, 0.01, 1)
+    , _sourcePath("sourcePath", "Source Path")
+    , _transferFunctionPath("transferFunctionPath", "Transfer Function Path")
+    , _raycaster(nullptr)
+    , _transferFunction(nullptr)
+    , _cache("cache", "Cache") {
+
+    glm::vec3 dimensions;
+    if (dictionary.getValue("Dimensions", dimensions)) {
```

Comment:
No need to do this:
`dictionary.getValue("Dimensions", _dimensions);`
will not change the value if the key doesn't exist or has the wrong type